### PR TITLE
Bug/ppna 4014 fix api inconsistency

### PR DESCRIPTION
**Pull request by pulina**
https://developers.piwik.pro/en/bug-ppna-4014-fix-api-inconsistency/custom_reports/object_management_api/object_management_api.html

Old version for comparison:

https://developers.piwik.pro/en/16.0/custom_reports/object_management_api/object_management_api.html

### CODE REVIEW

File: custom_reports/object_management_api/definitions/custom_dimension.yaml [hunk 1]
```diff
@@ -19,7 +20,16 @@ properties:
       - name
       - active
       - case_sensitive
+      - scope
+      - website_id
+      - slot
```
Comment (MarcinMaslany) [inline custom_reports/object_management_api/definitions/custom_dimension.yaml:25]: slot here, but is deleted below. 

It is read only so probably should be present, but definitely not required.

File: custom_reports/object_management_api/definitions/custom_dimension_create.yaml [hunk 2]
```diff
@@ -7,16 +7,17 @@ allOf:
         type: object
         required:
           - scope
-          - website_id
+          - slot
```
Comment (MarcinMaslany) [inline custom_reports/object_management_api/definitions/custom_dimension_create.yaml:10]: readonly field, so shouldn't be here probably

File: custom_reports/object_management_api/definitions/custom_dimension_details.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ allOf:
         type: object
         required:
           - scope
-          - website_id
+          - slot
```
Comment (MarcinMaslany) [inline custom_reports/object_management_api/definitions/custom_dimension_details.yaml:10]: Does `required` in openapi have different meaning? It appears in every case here, so it doesn't look like a mistake

File: custom_reports/object_management_api/definitions/goal.yaml [hunk 4]
```diff
@@ -1,15 +1,78 @@
...
-        required:
-          - website_id
-        properties:
-          website_id:
-            type: string
-            format: uuid
-            description: ID of the website.
-            example: dde2dc68-b6bb-45ee-9ac2-868b6181e33f
-  - $ref: '#/definitions/GoalDetails'
+      website_id:
```
Comment (MarcinMaslany) [inline custom_reports/object_management_api/definitions/goal.yaml:26]: don't we want to use refs? We do have         `$ref: '#/components/schemas/WebsiteId'`

Comment (pulina) [reply custom_reports/object_management_api/definitions/goal.yaml:26]: we have only `$ref: '#/parameters/WebsiteId'` but we could add components section in index.yaml.

File: custom_reports/object_management_api/definitions/custom_dimension_details.yaml (hunk 3 above)
Comment (MarcinMaslany) [reply custom_reports/object_management_api/definitions/custom_dimension_details.yaml:10]: yes, I was looking at wrong repo.

File: custom_reports/object_management_api/definitions/goal.yaml (hunk 4 above)
Comment (MarcinMaslany) [reply custom_reports/object_management_api/definitions/goal.yaml:26]: yes, I was looking at wrong repo.

**Review by MarcinMaslany — APPROVED** [review]

**Review by RomanSek — APPROVED** [review]

**Pull request merged**